documentation/swagger-extensions.md

```diff
@@ -531,6 +532,94 @@ parameters | [Array of Parameter Objects](https://github.com/OAI/OpenAPI-Specifi
   }
 ```
 
+##x-ms-mutability
+This extension offers insight to Autorest on how to generate code (mutability of the property of the model classes being generated). It doesn't alter the modeling of the actual payload that is sent on the wire.
+
+It is an array of strings with three possible values. The array cannot have repeatable values. Valid values are: **"create", "read", "update"**.
+
+Field Name | Description
+---|:---
+**create** | Indicates that the value of the property can be set while creating/initializing/constructing the object
+**read** | Indicates that the value of the property can be read
+**update** | Indicates that value of the property can be updated anytime(even after the object is created)
+
+###Rules:
+- When the extension is applied with all the three values; `"x-ms-mutability": ["create", "read", "update"]` (order of the values is not important) **OR** when this extension is not applied on a model property; it has the same effect in both the cases. Thus applying this extension with all the three values on all the settable properties is not required. This will ensure the spec is visibly cleaner.
+- When a property is modeled as `"readonly": true` then,
+  - if the x-ms-mutability extension is applied then it can **only have "read" value in the array**. 
+  - applying the extension as `"x-ms-mutability": ["read"]` or not applying it will have the same effect.
+- When the property is modeled as **`"readonly": false`** then,
+  - applying the extension as `"x-ms-mutability": ["read"]` is not allowed.
+  - applying the extension as `"x-ms-mutability": ["create", "read", "update"]` or not applying it will have the same effect.
+  - applying the extension with anyother **permissible valid combination** should be fine.
+- When this extension is applied on a collection (array, dictionary) then this will have effects on the mutability (adding/removing elements) of the collection. Mutabiility of the collection cannot be applied on its elements. The mutability of the element will be governed based on the mutability defined in the element's definition.
+
+Examples:
+- Mutability on a model definition
+```json5
+"definitions": {
+  "Resource": {
+    "description": "The Resource Model definition.",
+    "properties": {
+      "id": {
+        "readOnly": true,
+        "type": "string",
+        "description": "Resource Id",
+        "x-ms-mutability": ["read"]
+      },
+      "name": {
+        "type": "string",
```

Comment:
name is readOnly as well. NotificationHubs did not describe this as readOnly because of a bug in their implementation.